build_tools/bazel_to_cmake/bazel_to_cmake.py

```diff
@@ -535,8 +535,12 @@ def gentbl(self,
     td_file_block = self._convert_td_file_block(td_file)
     outs_block = self._convert_tbl_outs_block(tbl_outs)
 
-    self.converter.body += """iree_tablegen_library(
+    # Which kind of file to generate. We use the name as a convention here.
+    genfile_kind = "doc" if name.endswith("DocGen") else "library"
```

Comment:
Can we add this flag to the bazel rule as well rather than relying on naming conventions? I guess upstream we're using MLIR's tablegen rule, not our own, so maybe it's hard to do, but at least long term it would be nice to make this all more robust. MLIR uses CMake in OSS, so presumably we should be able to migrate whatever solution over to ther? Do they currently generate docs always? That seems gross.

I'd prefer to have the BUILD and CMake files look as similar as possible so bazel_to_cmake is a purely mechanical translation. To that end, we could wrap gentbl on the bazel side in macros for iree_tablegen_doc and iree_tablegen_library, so we have the same function split as in cmake.

Reply:
MLIR does generate docs at the moment. Doc generation registration is done via`add_mlir_doc`. We cannot reuse it directly given it has hard-wired a directory to collect all the generated docs and a master target to drive the generation.

Keeping things similar between Bazel and CMake is reasonable to me. I added a wrapper around `tblgen` with the same name on the CMake side, `iree_tblgen_doc`. Let me know how do you think. We can throw in another `iree_tblgen_library` similarly and change all existing `tblgen` uses to that.